modules/services/mpd-mpris.nix

```diff
@@ -6,85 +6,132 @@
 }:
 let
   cfg = config.services.mpd-mpris;
-
-  ignoreIfLocalMpd = value: if cfg.mpd.useLocal then null else value;
-
-  renderArg =
-    name: value:
-    if lib.isBool value && value then
-      "-${name}"
-    else if lib.isInt value then
-      "-${name} ${toString value}"
-    else if lib.isString value then
-      "-${name} ${lib.escapeShellArg value}"
-    else
-      "";
-
-  concatArgs = strings: lib.concatStringsSep " " (lib.filter (s: s != "") strings);
-
-  renderArgs = args: concatArgs (lib.mapAttrsToList renderArg args);
-
-  renderCmd = pkg: args: "${pkg}/bin/mpd-mpris ${renderArgs args}";
 in
 {
   meta.maintainers = [ lib.hm.maintainers.olmokramer ];
 
+  imports = [
+    (lib.mkRemovedOptionModule [ "services" "mpd-mpris" "mpd" "useLocal" ] ''
+      Just don't configure the network settings and it should automatically
+      connect to the local MPD server.
+    '')
+
+    (lib.mkRenamedOptionModule
+      [ "services" "mpd-mpris" "mpd" "network" ]
+      [ "services" "mpd-mpris" "settings" "network" ]
+    )
+
+    (lib.mkRenamedOptionModule
+      [ "services" "mpd-mpris" "mpd" "host" ]
+      [ "services" "mpd-mpris" "settings" "host" ]
+    )
+
+    (lib.mkRenamedOptionModule
+      [ "services" "mpd-mpris" "mpd" "port" ]
+      [ "services" "mpd-mpris" "settings" "port" ]
+    )
```

Comment:
We can use `lib.hm.deprecations` for this `mkSettingsRenamedOptionModules ` will allow you to shrink this down and reduce some duplication.